libtransmission/torrent.c

```diff
@@ -1131,6 +1132,72 @@ static tr_parse_result torrentParseImpl(tr_ctor const* ctor, tr_info* setmeInfo,
     return result;
 }
 
+static void tr_torrentAddDefaultTrackers(tr_torrent* tor)
+{
+    tr_list* l;
+    int tier = -1;
+    int i;
+    tr_tracker_info* trackers;
+    int numExistingTrackers = tor->info.trackerCount;
+    int numNewTrackers = tr_list_size(tor->session->defaultTrackers);
+    char* url;
+    bool duplicate;
+    bool changed = false;
+
+    if (!numNewTrackers || tor->info.isPrivate)
+    {
+        return;
+    }
+
+    // copy existing trackers
+    trackers = tr_new0(tr_tracker_info, numExistingTrackers + numNewTrackers);
+    for (i=0; i<numExistingTrackers; ++i)
+    {
+        trackers[i].tier = tor->info.trackers[i].tier;
+        trackers[i].announce = tr_strdup(tor->info.trackers[i].announce);
+        tier = MAX(tier, tor->info.trackers[i].tier);
+    }
+
+    // add the new ones
+    for (l=tor->session->defaultTrackers; l!=NULL; l=l->next)
+    {
+        url = (char*)l->data;
```

Comment:
the declaration for `url` could be moved to here, narrowing its scope. Also since this is passed walking through the session array it should be a const IMO. Also since `l->data` is a `void*` I don't think we need to explicitly cast it? So I think this would work:

`char const* url = l->data;`

Reply:
Changed.